Remotion/Data/DomainObjects/Queries/Configuration/Loader/QueryDefinitionFileLoader.cs

```diff
@@ -126,7 +126,7 @@ private QueryDefinition GetQueryDefinition (XmlNode queryNode, ConfigurationName
       if ((queryType is QueryType.CustomReadOnly or QueryType.CustomReadWrite) && collectionType != null)
         throw CreateQueryConfigurationException("A custom query '{0}' must not specify a collectionType.", queryID);
 
-      return new QueryDefinition(queryID, storageProviderDefinition, statement, queryType, collectionType);
+      return new QueryDefinition(queryID, storageProviderDefinition, statement, queryType, QueryStatementType.Text, collectionType);
```

Comment:
oops, missing feature: we need to add support for specifying the StatementType to the XML Schema (optional, to prevent breaking change) and then evaluate it here. we also need to add a new unit / integration test for the explicit stored procedures to document that both the old way to do things is unharmed and the new way works as well.